src/modules/FairFindPackage2.cmake.in

```diff
@@ -112,3 +112,8 @@ macro(find_package2 qualifier pkgname)
   set(CMAKE_PREFIX_PATH ${__old_cpp__})
   unset(__old_cpp__)
 endmacro()
+
+# Add oneself as PRIVATE package if used from the @PROJECT_NAME@ package
+if(@PROJECT_NAME@_FOUND)
+  set(PROJECT_PACKAGE_DEPENDENCIES ${PROJECT_PACKAGE_DEPENDENCIES} @PROJECT_NAME@)
```

Comment:
Why no `list(APPEND)`?

What do you think of adding this to demo1?
```cmake
if (NOT "FairCMakeModules" IN_LIST PROJECT_PACKAGE_DEPENDENCIES)
  message(FATAL_ERROR "FairCMakeModules missing from PROJECT_PACKAGE_DEPENDENCIES!")
if()
```

Reply:
Just copied from the function which also does not use list append. one reason for not using list append in the past was that not all cmake versions had it. Regarding content changes we should have some thought also into using some version check in each module, so it is clearer for which cmake versions they are intended.

Reply:
`list(APPEND)` doesn't even have a notice in the cmake docs. So it's likely 3.0?

+1 on having version checks in each module. After all we just "install" them here, others just "use" them.

Reply:
Added version check and `include_guard` as well as docs and test coverage for it.

> list(APPEND) doesn't even have a notice in the cmake docs. So it's likely 3.0?

Modernized to `list(APPEND ...)`.